crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -95,7 +95,8 @@ const INTERNAL_ALARMS_TOPIC: &str = "c8y-internal/alarms/";
 const C8Y_JSON_MQTT_EVENTS_TOPIC: &str = "c8y/event/events/create";
 const TEDGE_AGENT_LOG_DIR: &str = "tedge/agent";
 const CREATE_EVENT_SMARTREST_CODE: u16 = 400;
-const TEDGE_AGENT_HEALTH_TOPIC: &str = "tedge/health/tedge-agent";
+/// XXX: hardcoded topic root
+const TEDGE_AGENT_HEALTH_TOPIC: &str = "te/device/main/service/tedge-agent/status/health";
```

Comment:
> Now `tedge agent` also can run on a child device also right?

Not yet. The cumulocity mapper which runs on the main device only cares about the main device's agent. This constant is used here:

https://github.com/thin-edge/thin-edge.io/pull/2276/files#diff-08a20a8877f5e1e1bce1aac2c6fc0e1a63a36cc9311e7e60cb3aacacf3c8c57cR351-R352

which does filesystem operations on main device, so it can't be done for child devices as it is now.

Reply:
To not hardcode the MQTT root, rewrote this part to use EntityStore and MqttSchema to generate this topic in [5df9538](https://github.com/thin-edge/thin-edge.io/pull/2276/commits/5df95385746516786409c99c7f279cf028e082d2)

Reply:
> To not hardcode the MQTT root, rewrote this part to use EntityStore and MqttSchema to generate this topic in [5df9538](https://github.com/thin-edge/thin-edge.io/commit/5df95385746516786409c99c7f279cf028e082d2)

Beware, this commit is the root cause of the failing system tests.